Marlin/Marlin_main.cpp

```diff
@@ -4160,31 +4230,15 @@ inline void gcode_M226() {
     }
     SERIAL_EOL;
   }
+
   /**
    * M235 - Return processed external ADC value
    */
   inline void gcode_M235(uint8_t power) {
-
     if(code_seen('S')) {
       power = code_value();
     }
-
-    uint16_t num_samples = 0x0001 << power;
-    uint16_t i = 0;
-    uint32_t sample_sum = 0; // must be 32 bit unsigned int!
-    uint16_t sample_avg = 0;
-
-    // Value must be less than max sample power
-    // (This value was just taken from M234 for simplicity)
-    if(power > ADC_SAMPLE_POWER) {
-      power = ADC_SAMPLE_POWER;
-    }
-    // Take specified amount of readings
-    for(i = 0; i < num_samples; i++) {
-      sample_sum += EXT_ADC_READ_0;
-    }
-    // Take average of sample readings
-    sample_avg = sample_sum >> power;
```

Comment:
I don't understand why you've redefined M235 as M238? It seems to be the same thing, but M235 just returns what M238 does?

I think we should leave M235 as is and call gcode_M235() if it is needed.